crates/paimon/src/spec/types.rs

```diff
@@ -346,10 +429,13 @@ impl BooleanType {
 /// Data type of a fixed-length character string.
 ///
 /// Impl Reference: <https://github.com/apache/paimon/blob/release-0.8.2/paimon-common/src/main/java/org/apache/paimon/types/CharType.java>.
+#[serde_as]
 #[derive(Debug, Clone, PartialEq, Hash, Eq, Deserialize, Serialize)]
 #[serde(rename_all = "camelCase")]
 pub struct CharType {
+    #[serde_as(as = "DisplayFromStr")]
```

Comment:
Hi, I believe this is wrong since `CharType` should be serialized into `CHAR(n) NOT NULL`.

The `DisplayFromStr` should be added upon `CharType` instead of field instead. Also, `CharType` should implement `FromStr` which can parse a `CharType` from given string `CHAR(n) NOT NULL`.